File: packages/jsii-pacmak/test/expected.jsii-calc-base-of-base/java/src/main/java/software/amazon/jsii/tests/calculator/baseofbase/IVeryBaseInterface.java
package software.amazon.jsii.tests.calculator.baseofbase;

@javax.annotation.Generated(value = "jsii-pacmak")
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.baseofbase.$Module.class, fqn = "@scope/jsii-calc-base-of-base.IVeryBaseInterface")
@software.amazon.jsii.Jsii.Proxy(IVeryBaseInterface.Jsii$Proxy.class)
public interface IVeryBaseInterface extends software.amazon.jsii.JsiiSerializable {

    void foo();

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    final static class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements software.amazon.jsii.tests.calculator.baseofbase.IVeryBaseInterface {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(objRef);
        }

        @Override
        public void foo() {
            this.jsiiCall("foo", Void.class);
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc-base-of-base/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/baseofbase/IVeryBaseInterface.kt
package software.amazon.jsii.tests.kotlin.calculator.baseofbase

@javax.annotation.Generated("jsii-pacmak")
interface IVeryBaseInterface : software.amazon.jsii.JsiiSerializable {
    fun foo()

    class `Jsii$Proxy` : software.amazon.jsii.JsiiObject, software.amazon.jsii.tests.kotlin.calculator.baseofbase.IVeryBaseInterface {
        protected constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef) {
        }

        override fun foo() {
            this.jsiiCall("foo", kotlin.Unit::class.java) ?: error("Method 'foo()' returned null value")
        }
    }
}
